std/meta.d

```diff
@@ -1094,42 +1094,23 @@ template templateOr(Preds...)
     }
 }
 
+import std.traits : isIterable;
+
 /**
- * Converts an input range $(D range) to an alias sequence.
+ * Converts any foreach-iterable (e.g. an input range) $(D range) to an alias sequence.
  */
-template aliasSeqOf(alias range)
+template aliasSeqOf(alias iter)
+if (isIterable!(typeof(iter)))
 {
-    import std.traits : isArray, isNarrowString;
+    import std.traits : isNarrowString;
+    import std.array : array;
 
-    alias ArrT = typeof(range);
-    static if (isArray!ArrT && !isNarrowString!ArrT)
+    struct Impl
     {
-        static if (range.length == 0)
-        {
-            alias aliasSeqOf = AliasSeq!();
-        }
-        else static if (range.length == 1)
-        {
-            alias aliasSeqOf = AliasSeq!(range[0]);
-        }
-        else
-        {
-            alias aliasSeqOf = AliasSeq!(aliasSeqOf!(range[0 .. $/2]), aliasSeqOf!(range[$/2 .. $]));
-        }
-    }
-    else
-    {
-        import std.range.primitives : isInputRange;
-        static if (isInputRange!ArrT)
-        {
-            import std.array : array;
-            alias aliasSeqOf = aliasSeqOf!(array(range));
-        }
-        else
-        {
-            static assert(false, "Cannot transform range of type " ~ ArrT.stringof ~ " into a AliasSeq.");
-        }
+        static foreach (size_t i, el; iter.array)
```

Comment:
don't use `.array` please.
instead introduce a counter variable into the parent-scope.